src/barcode_query_app/app_async.rs

```diff
@@ -0,0 +1,20 @@
+use crate::barcode_query::model;
+use crate::web::state::SharedState;
+use std::ops::Deref;
+use std::time::Duration;
+
+pub struct BarcodeAppAsync;
```

Comment:
Why do we need `Async` in the name? Do we have the `Sync` version? If not I don't think we should put the Async in the name.

Reply:
yes, there is an old Sync version. The naming just for clarity at the moment and will be removed when the async impl is complete